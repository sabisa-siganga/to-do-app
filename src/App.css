/* react-bootstrap bootstrap link */
@import 'bootstrap/dist/css/bootstrap.min.css';

/* styling of the todo container  */
.todo-container {
  text-align: center;
  background: #f3f0f0;
  box-shadow: 1px 1px 37px #a7a4a4;
  max-width: 600px;
  height: 400px;
  margin: 0 auto;
  padding-top: 20px;
  margin-top: 30px;
  margin-bottom: 30px;
  border-radius: 10px;



  /* styling of the input field container */
  .input-field {
    display: flex;
    flex-direction: column;
    row-gap: 10px;
    margin-bottom: 20px;
    align-items: center;

    /* styling the label of the input field */
    label {
      font-size: 20px;
      font-weight: 400;
    }

    /* styling of the input field  */
    input {
      width: 200px;
      height: 37px;
      border: 2px solid #541554;
      border-radius: 5px;
      padding: 0 10px;
      font-size: 15px;
    }
  }


  /* styling of the buttons container */
  .buttons {
    display: flex;
    flex-direction: column;
    row-gap: 10px;
    align-items: center;

  }

  /* styling of the buttons*/
  .btn {
    width: 111px;
    height: 40px;
    background: #541554;
    color: #fff;
    border-radius: 5px;
    border: 2px solid #fff;
    font-size: 15px;

  }

}