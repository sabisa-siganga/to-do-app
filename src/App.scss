/* react-bootstrap bootstrap link */
@import "bootstrap/dist/css/bootstrap.min.css";

/* google fonts link */
@import url("https://fonts.googleapis.com/css2?family=Roboto:ital,wght@0,100;0,300;0,400;0,500;0,700;0,900;1,300;1,400&display=swap");

/* styling of the web page body  */
body {
  font-family: "Roboto", sans-serif;
}

/* styling of the todo container  */
.todo-container {
  background: #f3f0f0;
  box-shadow: 1px 1px 37px #a7a4a4;
  max-width: 600px;
  height: 500px;
  margin: 0 auto;
  padding-top: 20px;
  margin-top: 30px;
  margin-bottom: 30px;
  border-radius: 10px;

  /* styling of the form */
  .form-info {
    display: flex;
    justify-content: center;

    /* styling the title */
    .title {
      font-size: 25px;
      font-weight: 400;
      color: #541554;
    }
    /* styling of the input field container */
    .input-container {
      display: flex;
      flex-direction: column;
      row-gap: 10px;
      align-items: center;

      /* styling of the input field  */
      input {
        width: 400px;
        height: 37px;
        border: 2px solid #541554;
        border-radius: 5px 0 0 5px;
        padding: 0 10px;
        outline: none;
      }
    }

    /* styling of the add button  */
    .add-button {
      width: 45px;
      height: 37px;
      background: #541554;
      color: #fff;
      border-radius: 0 5px 5px 0;
      border: 2px solid #541554;
      font-size: 15px;
      display: flex;
      justify-content: center;
      align-items: center;
      padding: 0;
    }
  }

  /* styling of the input and add button container  */
  .input-add-btn {
    margin-bottom: 35px;
    padding: 0 70px;
  }
  /* styling of the todo title */
  .heading {
    padding: 0 65px;
    font-weight: 400;
  }
}
