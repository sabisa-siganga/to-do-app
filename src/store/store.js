// importing the configureStore from reduxjs toolkit
//  to configure the store in order to store the state
import { configureStore } from "@reduxjs/toolkit";

// importing the bankReducer
import todoReducer from "./todo";

export default configureStore({
  // reducer
  reducer: {
    todo: todoReducer,
  },
});
