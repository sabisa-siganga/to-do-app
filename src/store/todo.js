// importing createSlice from reduxjs/toolkit to create a slice
import { createSlice } from "@reduxjs/toolkit";

export const todoSlice = createSlice({
  name: "todo",

  // initializing state
  initialState: {
    nextId: 2,
    data: {
      1: {
        content: "Content 1",
        completed: false,
      },
    },
  },

  reducers: {
    /**
     * Adding a todo item to the state
     * @param {object} state
     * @param {object} action
     */
    addTodo: (state, action) => {
      state.data[state.nextId] = {
        content: action.payload.content,
        completed: false,
      };
      state.nextId += 1;
    },

    /**
     * Editing a todo item in the state
     * @param {object} state
     * @param {object} action
     */
    editTodo: (state, action) => {
      const todo = state.data[action.payload.id];

      state.data[action.payload.id] = {
        completed: todo.completed,
        content: action.payload.content,
      };
    },

    /**
     * deleting a todo item from the state
     * @param {object} state
     * @param {object} action
     */
    deleteTodo: (state, action) => {
      delete state.data[action.payload.id];
    },

    /**
     * checking a todo item in the state once completed
     * @param {object} state
     * @param {object} action
     */
    completeTodo: (state, action) => {
      const complete = state.data[action.payload.id];
      state.data[action.payload.id] = {
        completed: !complete.completed,
        content: complete.content,
      };
    },
  },
});

// exporting the addTodo, editTodo, deleteTodo and completeTodo actions
export const { addTodo, editTodo, deleteTodo, completeTodo } =
  todoSlice.actions;

export default todoSlice.reducer;
