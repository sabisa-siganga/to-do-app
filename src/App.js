import "./App.css";
import { Button, Container } from "react-bootstrap";

function App() {
  return (
    <Container className="todo-container">
      <form>
        <div className="input-field">
          <label>To-do List</label>
          <input placeholder="Please enter a to-do" />
        </div>

        <div className="todos-list">
          <ul>
            <li>list</li>
          </ul>
        </div>

        <div className="buttons">
          <Button className="btns">
            <span class="material-symbols-outlined">edit</span>
          </Button>
          <Button className="btns">
            <span class="material-symbols-outlined">delete</span>
          </Button>
          <Button className="btns">
            <span class="material-symbols-outlined">done</span>
          </Button>
        </div>
      </form>
    </Container>
  );
}

export default App;
