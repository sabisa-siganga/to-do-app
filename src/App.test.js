import { render, screen } from "@testing-library/react";
import App from "./App";
import renderer from "react-test-renderer";
import { Provider } from "react-redux";
import store from "./store/store";

// creating a test suited for the todo component
describe("Renders: <App />", () => {
  let component;

  // recreating app component on before each test
  beforeEach(() => {
    component = (
      <Provider store={store}>
        <App />
      </Provider>
    );
  });

  // Describing the test
  it("renders base component", () => {
    render(component);

    const linkElement = screen.getByText(/Todo List/i);
    expect(linkElement).toBeInTheDocument();
  });

  // taking the snapshot of the component
  it("taking a snapshot", () => {
    const tree = renderer.create(component, {}).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
