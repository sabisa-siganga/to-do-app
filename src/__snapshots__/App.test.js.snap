// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renders: <App /> taking a snapshot 1`] = `
<div
  className="todo-container container"
>
  <h2
    className="title mb-5 text-center"
  >
    Todo List
  </h2>
  <div
    className="input-add-btn"
  >
    <form
      className="form-info"
      onSubmit={[Function]}
    >
      <div
        className="input-container"
      >
        <input
          onChange={[Function]}
          placeholder="Please enter a to-do"
          required={true}
          value=""
        />
      </div>
      <div
        className="add-btn"
      >
        <button
          className="add-button btn btn-primary"
          disabled={false}
          type="submit"
        >
          <span
            className="material-symbols-outlined py-1 "
          >
            add
          </span>
        </button>
      </div>
    </form>
  </div>
  <div
    className="heading mb-3"
  >
    Todos:
  </div>
  <div
    className="todo-list"
  >
    <div
      className="list-btns-cont px-5"
    >
      <div
        className="todo-content "
      >
        Content 1
      </div>
      <div
        className="buttons"
      >
        <button
          className="btns px-0 py-0 btn btn-primary"
          disabled={false}
          onClick={[Function]}
          type="button"
        >
          <span
            className="material-symbols-outlined py-1"
          >
            edit
          </span>
        </button>
        <button
          className="btns px-0 py-0 btn btn-primary"
          disabled={false}
          onClick={[Function]}
          type="button"
        >
          <span
            className="material-symbols-outlined py-1"
          >
            delete
          </span>
        </button>
        <button
          className="btns px-0 py-0 btn btn-primary"
          disabled={false}
          onClick={[Function]}
          type="button"
        >
          <span
            className="material-symbols-outlined py-1"
          >
            done
          </span>
        </button>
      </div>
    </div>
  </div>
</div>
`;
